### PR TITLE
[Sparse] Reduce peak memory of SpSpAdd.

### PR DESCRIPTION
**Pull request by czkkkkkk**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: dgl_sparse/src/elemenwise_op.cc [hunk 1]
```diff
@@ -24,7 +24,10 @@ c10::intrusive_ptr<SparseMatrix> SpSpAdd(
   ElementwiseOpSanityCheck(A, B);
   auto torch_A = COOToTorchCOO(A->COOPtr(), A->value());
   auto torch_B = COOToTorchCOO(B->COOPtr(), B->value());
-  auto sum = (torch_A + torch_B).coalesce();
+  auto sum = torch_A + torch_B;
+  torch_A = torch::Tensor();
```
Comment (frozenbugs) [inline dgl_sparse/src/elemenwise_op.cc:28]: This is a little bit ugly.

File: dgl_sparse/src/elemenwise_op.cc [hunk 2]
```diff
@@ -24,7 +24,10 @@ c10::intrusive_ptr<SparseMatrix> SpSpAdd(
   ElementwiseOpSanityCheck(A, B);
   auto torch_A = COOToTorchCOO(A->COOPtr(), A->value());
   auto torch_B = COOToTorchCOO(B->COOPtr(), B->value());
-  auto sum = (torch_A + torch_B).coalesce();
+  auto sum = torch_A + torch_B;
```
Comment (frozenbugs) [inline dgl_sparse/src/elemenwise_op.cc:27]: This is not a long term solution, let's file a bug to discuss the better solution in the future for avoiding memory copy when converting from DGL sparse to torch sparse.

File: dgl_sparse/src/elemenwise_op.cc (hunk 1 above)
Comment (frozenbugs) [reply dgl_sparse/src/elemenwise_op.cc:28]: @Rhett-Ying

Comment (Rhett-Ying) [reply dgl_sparse/src/elemenwise_op.cc:28]: how about wrap `torch_A/B` into a `{}` then they'll be destructed automatically once leaves the domain.

Comment (Rhett-Ying) [reply dgl_sparse/src/elemenwise_op.cc:28]: `lambda` could be another alternative.

**Review by Rhett-Ying — APPROVED** [review]

File: dgl_sparse/src/elemenwise_op.cc [hunk 3]
```diff
@@ -22,9 +22,14 @@ c10::intrusive_ptr<SparseMatrix> SpSpAdd(
     const c10::intrusive_ptr<SparseMatrix>& A,
     const c10::intrusive_ptr<SparseMatrix>& B) {
   ElementwiseOpSanityCheck(A, B);
-  auto torch_A = COOToTorchCOO(A->COOPtr(), A->value());
-  auto torch_B = COOToTorchCOO(B->COOPtr(), B->value());
-  auto sum = (torch_A + torch_B).coalesce();
+  torch::Tensor sum;
```
Comment (frozenbugs) [inline dgl_sparse/src/elemenwise_op.cc:25]: Add 
`// TODO(#1234)`

**Pull request merged**